### DISCUSSION TITLE
Call javascipt window function from gotenberg.

### DISCUSSION THREAD

**Discussion by Fadil1710** [gotenberg/gotenberg#936] [Q&A]
Am having a JavaScript window function in my html page , I want to trigger the window function from my c# backend before printing the page. tried using waitForExpression , it didn't helped

C# sample code:
var options = new
{
    waitForExpression = "window.manipulateDOM()"
}

UI sample code:
    <script>
        window.manipulateDOM = function() {
            setTimeout(() => {
                document.getElementsByTagName('body')[0].innerHTML = "Script Loaded";
                return true;
            }, 5000);
        };
    </script>
    
I want gotenberg to wait for 5 seconds and only after the manipulateDOM method is executed, the pdf should be generated , I also tried waitForExpression with a simple boolean property , it didn't worked as well.

am sending formData as multipart/form-data media type

**Comment by gulien** [comment]
Hello @Fadil1710,

You may try:

```js
<script type="application/javascript">
    var globalVar = 'notReady'

    const delay = ms => new Promise(res => setTimeout(res, ms))
    delay(5000).then(() => {
        document.getElementsByTagName('body')[0].innerHTML = "Script Loaded";
        window.globalVar = 'ready'
    })
</script>
```

With the form field:

```
curl \
--request POST http://localhost:3000/forms/chromium/convert/url \
--form url=https://my.url \
--form 'waitForExpression=window.globalVar === '\''ready'\''' \
-o my.pdf
```
